ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala

```diff
@@ -40,36 +46,95 @@ import scala.util.control.NoStackTrace
   * H2O AutoML algorithm exposed via Spark ML pipelines.
   */
 class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
-  with H2OAlgoCommonUtils with DefaultParamsWritable with H2OAutoMLParams {
+  with H2OAlgoCommonUtils with DefaultParamsWritable with H2OAutoMLParams with RestCommunication {
 
   // Override default values
   setDefault(nfolds, 5)
 
-  private lazy val spark = SparkSessionUtils.active
-
   def this() = this(Identifiable.randomUID(classOf[H2OAutoML].getSimpleName))
 
-  private var amlOption: Option[AutoML] = None
+  private var amlKeyOption: Option[String] = None
+
+  private def getInputSpec(trainKey: String, validKey: Option[String]): Map[String, Any] = {
+    Map(
+      "response_column" -> getLabelCol(),
+      "fold_column" -> getFoldCol(),
+      "weights_column" -> getWeightCol(),
+      "ignored_columns" -> getIgnoredCols(),
+      "sort_metric" -> getSortMetric(),
+      "training_frame" -> trainKey
+    ) ++ validKey.map{ key => Map("validation_frame" -> key) }.getOrElse(Map())
+  }
 
-  override def fit(dataset: Dataset[_]): H2OMOJOModel = {
-    val spec = new AutoMLBuildSpec
-    amlOption = None
-    val (trainKey, validKey, internalFeatureCols) = prepareDatasetForFitting(dataset)
-    spec.input_spec.training_frame = DKV.getGet[Frame](trainKey)._key
-    spec.input_spec.validation_frame = validKey.map(DKV.getGet[Frame](_)._key).orNull
+  private def getBuildModels(): Map[String, Any] = {
+    val monotoneConstraints = getMonotoneConstraints()
+    val algoParameters = if (monotoneConstraints != null && monotoneConstraints.nonEmpty) {
+      Map("monotone_constrains" -> monotoneConstraints)
+    } else {
+      Map()
+    }
+    Map(
+      "include_algos" -> determineIncludedAlgos(),
+      "exclude_algos" -> null
+    ) ++ (if (algoParameters.nonEmpty) Map("algo_parameters" -> algoParameters) else Map())
```

Comment:
Is the nesting of maps defined by H2O rest API?

Reply:
Yup, actually it is required in this case